importer/wdi_importer.py

```diff
@@ -1,63 +1,922 @@
+                    else:
+                        global_cat[indicator]['limitations'] = ''
+                if column_number == 12:
+                    if cell.value:
+                        global_cat[indicator]['sourcenotes'] = cell.value
+                    else:
+                        global_cat[indicator]['sourcenotes'] = ''
+                if column_number == 13:
+                    if cell.value:
+                        global_cat[indicator]['comments'] = cell.value
+                    else:
+                        global_cat[indicator]['comments'] = ''
+                if column_number == 14:
+                    global_cat[indicator]['source'] = cell.value
+                if column_number == 15:
+                    if cell.value:
+                        global_cat[indicator]['concept'] = cell.value
+                    else:
+                        global_cat[indicator]['concept'] = ''
+                if column_number == 17:
+                    if cell.value:
+                        global_cat[indicator]['sourcelinks'] = cell.value
+                    else:
+                        global_cat[indicator]['sourcelinks'] = ''
+                if column_number == 18:
+                    if cell.value:
+                        global_cat[indicator]['weblinks'] = cell.value
+                    else:
+                        global_cat[indicator]['weblinks'] = ''
+                global_cat[indicator]['saved'] = False
+
+        column_number = 0
+
+    category_vars = {}  # categories and their corresponding variables
+
+    for key, value in global_cat.items():
+        if category_vars.get(value['category'], 0):
+            category_vars[value['category']].append(key)
+        else:
+            category_vars[value['category']] = []
+            category_vars[value['category']].append(key)
+
+    existing_categories = DatasetCategory.objects.values('name')
+    existing_categories_list = []
+
+    for each in existing_categories:
+        existing_categories_list.append(each['name'])
+
+    if wdi_category_name_in_db not in existing_categories_list:
+        the_category = DatasetCategory(name=wdi_category_name_in_db)
+        the_category.save()
+        backup_sql_file.write("INSERT INTO %s (name, created_at, updated_at) VALUES ('%s', '%s', '%s');\n" %
+                              (DatasetCategory._meta.db_table, wdi_category_name_in_db.replace("'", "\\'"),
+                               the_category.created_at.strftime('%Y-%m-%d %H:%M:%S'),
+                               the_category.updated_at.strftime('%Y-%m-%d %H:%M:%S')))
+        logger.info("Inserting a category %s." % wdi_category_name_in_db.encode('utf8'))
+
+    else:
+        the_category = DatasetCategory.objects.get(name=wdi_category_name_in_db)
+
+    existing_subcategories = DatasetSubcategory.objects.filter(fk_dst_cat_id=the_category.pk).values('name')
+    existing_subcategories_list = []
+
+    for each in existing_subcategories:
```

Comment:
When constructing data used solely for existence testing, it's generally better to use set() or dict() instead-- O(1) instead of O(n)